boringtun/src/device/drop_privileges.rs

```diff
@@ -5,23 +5,35 @@ use crate::device::errno_str;
 use crate::device::Error;
 use libc::*;
 
+#[cfg(target_os = "macos")]
+use nix::unistd::User;
+
 pub fn get_saved_ids() -> Result<(uid_t, gid_t), Error> {
     // Get the user name of the sudoer
-    let uname = unsafe { getlogin() };
-    if uname.is_null() {
-        return Err(Error::DropPrivileges("NULL from getlogin".to_owned()));
+    #[cfg(target_os = "macos")]
+    {
+        let uname: &'static str = env!("USER");
```

Comment:
env!() retrieves the environment variable at compile time, is it expected? If I compile the code in one user, and use the binary in another user(or machine), the behavior is incorrect.

Reply:
I'll open up a PR after I ear breakfast.

Reply:
Also, next time, it's much easier to track this if you create an issue to track the problem.

Reply:
Fixed